File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    answer = 0 
    dic = {}
    L = len(want)
    for i in range(L):
        dic[want[i]] = number[i]
    
    l = len(discount)
    for i in range(l):
        if i+10 > l:
            continue
        new_dic = dic.copy()
        nbp = False
        for j in range(i,i+10):
            now = discount[j]
            if now not in dic:
                continue
            new_dic[now] -= 1

        for v in new_dic.values():
            if v > 0:
                nbp = True
                break
        
        if not nbp:
            answer += 1
            
            
    
    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.cpp
#include <string>
#include <vector>
#include <iostream>
#include <map>
using namespace std;


// map 복사하는 방법
int solution(vector<string> want, vector<int> number, vector<string> discount) {
    int answer = 0;
    map<string, int> dic;
    for (int i = 0; i < want.size(); i++) {
        dic[want[i]] = number[i];
    }

    int l = discount.size();
    for (int i = 0; i <= l - 10; i++) {
        map<string, int> tmp(dic);
        bool tf = true;

        for (int j = i; j < i + 10; j++) {
            tmp[discount[j]]--;
        }

        for (const auto& c : tmp) {
            if (c.second > 0) { 
                tf = false;
                break;
            }
        }
        if (tf) {
            answer++;
        }
    }

    return answer;
}
